atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/external/SchemalessStreamStoreDeleter.java

```diff
@@ -0,0 +1,127 @@
+        if (streamIds.isEmpty()) {
+            log.debug("deleteStreams() was called with no identifiers, so we are returning.");
+            return;
+        }
+
+        // Safe, because these are longs.
+        log.info("Now attempting to delete streams from the {} stream store, with identifiers {}",
+                LoggingArgs.tableRef(getTableReference(StreamTableType.VALUE)),
+                SafeArg.of("streamIds", streamIds));
+
+        Map<GenericStreamIdentifier, StreamMetadata> metadataInDb = metadataReader.readMetadata(tx, streamIds);
+        StreamStoreCellDeletion deletion = findCellsToPropagateDeletes(metadataInDb);
+        executeCellDeletion(tx, deletion);
+    }
+
+    private StreamStoreCellDeletion findCellsToPropagateDeletes(
+            Map<GenericStreamIdentifier, StreamMetadata> streamIdentifierToMetadata) {
+        ImmutableStreamStoreCellDeletion.Builder builder = ImmutableStreamStoreCellDeletion.builder();
+
+        for (Map.Entry<GenericStreamIdentifier, StreamMetadata> entry : streamIdentifierToMetadata.entrySet()) {
+            GenericStreamIdentifier streamId = entry.getKey();
+            StreamMetadata streamMetadata = entry.getValue();
+
+            builder.addAllValueTableCellsToDelete(cellCreator.constructValueTableCellSet(
+                    streamId, getNumberOfBlocksFromMetadata(entry.getValue())));
+            builder.addMetadataTableCellsToDelete(cellCreator.constructMetadataTableCell(streamId));
+            builder.addHashTableCellsToDelete(cellCreator.constructHashTableCell(streamId, streamMetadata.getHash()));
+        }
+
+        return builder.build();
+    }
+
+    private void executeCellDeletion(Transaction tx, StreamStoreCellDeletion deletion) {
+        transactionallyDeleteCells(tx, StreamTableType.VALUE, deletion.valueTableCellsToDelete());
+        transactionallyDeleteCells(tx, StreamTableType.METADATA, deletion.metadataTableCellsToDelete());
+        transactionallyDeleteCells(tx, StreamTableType.HASH, deletion.hashTableCellsToDelete());
+
+        log.info("Cells deleted were {} from the {} table, {} from the {} table and {} from the {} table.",
+                UnsafeArg.of("valueCells", deletion.valueTableCellsToDelete()),
+                LoggingArgs.tableRef("valueTable", getTableReference(StreamTableType.VALUE)),
+                UnsafeArg.of("metadataCells", deletion.metadataTableCellsToDelete()),
+                LoggingArgs.tableRef("metadataTable", getTableReference(StreamTableType.METADATA)),
+                UnsafeArg.of("hashCells", deletion.hashTableCellsToDelete()),
+                LoggingArgs.tableRef("hashTable", getTableReference(StreamTableType.HASH)));
+    }
+
+    private void transactionallyDeleteCells(Transaction tx, StreamTableType streamTableType, Set<Cell> cellsToDelete) {
+        tx.delete(getTableReference(streamTableType), cellsToDelete);
+    }
+
+    @VisibleForTesting
```

Comment:
General tip: I guess you added the methods as `@VisibleForTesting` because it didn't feel necessary/appropriate for this class to expose the methods publicly, which is a hint that the logic belongs elsewhere.

Consider inverting control here and having a public method `StreamTableType#getTableName(namespace, streamStoreShortName)` (and similarly for the metadata)?

Reply:
Yep, that could work. The methods are tricky enough (well, maybe not `getTableReference`...) that I want them tested as well (as opposed to simply leaving them private). I though they were quite small, but it does make sense for gTR to be in StreamTableType (and same for the stream metadata)